src/main/java/io/mosip/digitalcard/service/impl/UinCardGeneratorImpl.java

```diff
@@ -0,0 +1,127 @@
+package io.mosip.digitalcard.service.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.mosip.digitalcard.constant.ApiName;
+import io.mosip.digitalcard.constant.UinCardType;
+import io.mosip.digitalcard.dto.ErrorDTO;
+import io.mosip.digitalcard.dto.PDFSignatureRequestDto;
+import io.mosip.digitalcard.dto.SignatureResponseDto;
+import io.mosip.digitalcard.exception.DigitalCardServiceException;
+import io.mosip.digitalcard.service.UinCardGenerator;
+import io.mosip.digitalcard.util.DigitalCardRepoLogger;
+import io.mosip.digitalcard.util.RestClient;
+import io.mosip.kernel.core.exception.ServiceError;
+import io.mosip.kernel.core.http.RequestWrapper;
+import io.mosip.kernel.core.http.ResponseWrapper;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.kernel.core.pdfgenerator.exception.PDFGeneratorException;
+import io.mosip.kernel.core.pdfgenerator.spi.PDFGenerator;
+import io.mosip.kernel.core.util.DateUtils;
+import io.mosip.kernel.pdfgenerator.itext.constant.PDFGeneratorExceptionCodeConstant;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.env.Environment;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Class UinCardGeneratorImpl.
+ * 
+ * @author Dhanendra.
+ */
+@Component
+public class UinCardGeneratorImpl implements UinCardGenerator<byte[]> {
```

Comment:
My understand is we are using the below layers for generating the PDF card.

DigitalCardController
DigitalCardService
PDFCardService
UinCardGenerator
PDFGenerator

Can we avoid the UinCardGenerator layer and directly include those logic into the PDFCardService itself ? Since we are copying the code redirectly, I don't see relevance in maintaining many layers

Reply:
taken care